libraries/AP_CheckFirmware/AP_CheckFirmware.h

```diff
@@ -43,24 +47,40 @@ enum class check_fw_result_t : uint8_t {
  are filled in by set_app_descriptor() in the waf build
  */
 struct app_descriptor {
+#if AP_SIGNED_FIRMWARE
+    uint8_t sig[8] = { 0x41, 0xa3, 0xe5, 0xf2, 0x65, 0x69, 0x92, 0x07 };
+#else
     uint8_t sig[8] = { 0x40, 0xa2, 0xe4, 0xf1, 0x64, 0x68, 0x91, 0x06 };
+#endif
     // crc1 is the crc32 from firmware start to start of image_crc1
     uint32_t image_crc1 = 0;
     // crc2 is the crc32 from the start of version_major to the end of the firmware
     uint32_t image_crc2 = 0;
     // total size of firmware image in bytes
     uint32_t image_size = 0;
     uint32_t git_hash = 0;
+
+#if AP_SIGNED_FIRMWARE
+    // firmware signature
+    uint32_t signature_length = 0;
+    uint8_t signature[72] = {};
+#endif
     // software version number
     uint8_t  version_major = APP_FW_MAJOR;
     uint8_t version_minor = APP_FW_MINOR;
     // APJ_BOARD_ID (hardware version). This is also used in CAN NodeInfo
     // with high byte in HardwareVersion.major and low byte in HardwareVersion.minor
     uint16_t  board_id = APJ_BOARD_ID;
     uint8_t reserved[8] = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
+
```

Comment:
lets make board_id 32 bit when signing is enabled. I have a feeling we will end up needing more IDs
the bootloader protocol already supports 32 bit

Reply:
If the boot loader relies on the board id in the apj file being correct (i.e. its not in the image), could someone not create a modified apj for a board that actually contains a signed image for a different board? Worst case this could allow a malicious user to cause a board to fail, am I right?